slsReceiverSoftware/src/slsReceiverTCPIPInterface.cpp

```diff
@@ -682,1580 +547,793 @@ int slsReceiverTCPIPInterface::set_roi(sls::ServerInterface2 &socket) {
 	if (myDetectorType == EIGER || myDetectorType == JUNGFRAU)
 		functionNotImplemented();
 
-	// base object not null
-	else if (receiver == nullptr)
-		NullObjectError(ret, mess);
-	else {
-		if (VerifyLockAndIdle(ret, mess, fnum) == OK)
-			ret = receiver->setROI(arg);
-	}
-	return socket.sendResult(true, ret, nullptr, 0, mess);
+	VerifyIdle(socket);
+	if (impl()->setROI(arg) == FAIL)
+		throw RuntimeError("Could not set ROI");
+	return socket.sendData(OK);
 }
 
-
-
-int slsReceiverTCPIPInterface::setup_udp(sls::ServerInterface2 &socket){
+int slsReceiverTCPIPInterface::setup_udp(Interface &socket){
 	ret = OK;
-	memset(mess, 0, sizeof(mess));
 	char args[5][MAX_STR_LENGTH]{};
 	char retvals[2][MAX_STR_LENGTH]{};
-
-	// get args, return if socket crashed, ret is fail if receiver is not null
-	socket.receiveArg(ret, mess, args, sizeof(args));
-	if (receiver == nullptr){
-		NullObjectError(ret, mess);
-		return FAIL;
+	socket.receiveArg(args);
+	VerifyIdle(socket);
+
+	//setup interfaces count
+	int numInterfaces = atoi(args[0]) > 1 ? 2 : 1;
+	char* ip1 = args[1];
+	char* ip2 = args[2];
+	uint32_t port1 = atoi(args[3]);
+	uint32_t port2 = atoi(args[4]);
+
+	// 1st interface
+	impl()->setUDPPortNumber(port1);
+	if (myDetectorType == EIGER) {
+		impl()->setUDPPortNumber2(port2);
+	}
+	FILE_LOG(logINFO) << "Receiver UDP IP: " << ip1;
+	// get eth
+	std::string temp = sls::IpToInterfaceName(ip1);
+	if (temp == "none"){
+		throw RuntimeError("Failed to get ethernet interface or IP");
+	} else {
+		char eth[MAX_STR_LENGTH]{};
+		strcpy(eth, temp.c_str());
+		// if there is a dot in eth name
+		if (strchr(eth, '.') != nullptr) {
+			strcpy(eth, "");
+			ret = FAIL;
```

Comment:
This was there previously, but I now think "ret = FAIL" should be removed here and mac address should always be checked even if eth is blank. Then you can remove the following if (ret!=FAIL) for mac address and sendResult like the other functions. The same for the second part for jungfrau.

Reply:
I'll trust you on this one